File: clase4/src/publicador.py
import rospy
from std_msgs.msg import String
from geometry_msgs.msg import Twist


def main():
    rospy.init_node('publicador', anonymous=True)
    

    # Create a publisher for Twist messages
    pub_twist = rospy.Publisher('cmd_vel', Twist, queue_size=10)
    
    rate = rospy.Rate(10)  # 10 Hz
    
    while not rospy.is_shutdown():

        # Publish a Twist message
        twist_msg = Twist()
        twist_msg.linear.x = 1.0
        twist_msg.angular.z = 0.5
        pub_twist.publish(twist_msg)
        
        rate.sleep()

File: test_package/src/publicador.cpp
#include "ros/ros.h"
#include "geometry_msgs/Twist.h"
#include <sstream>


int main(int argc, char **argv)
{

  ros::init(argc, argv, "talker");

  // Objete que me permite manejar el nodo
  ros::NodeHandle n;
  //creo un objeto publicador y lo enlazo con el nodo
  ros::Publisher pub_cmd_vel = n.advertise<geometry_msgs::Twist>("/turtle1/cmd_vel", 10);

  //crear un objeto para controlar la frecuencia de publicación
  ros::Rate loop_rate(10);

  geometry_msgs::Twist msg_cmd_vel;
  msg_cmd_vel.angular.z = 0.5;
  msg_cmd_vel.linear.x = 0.5;
  while (ros::ok())
  {

    pub_cmd_vel.publish(msg_cmd_vel);
    ROS_INFO("Enviando comando de velocidad : x [%.3f]  angular [%.3f]", 
      msg_cmd_vel.linear.x, msg_cmd_vel.angular.z);
    loop_rate.sleep();
  }

  return 0;
}